### PR TITLE
Clean `docker run -e` reference docs

### PR DESCRIPTION
**Pull request by joaofnfernandes**
Simplified the docs on how to set environment variables in a
container. Makes it clear that you have three options, and how
to use them.

Relates to docker/docker.github.io#2355
/cc @mstanleyjones 

Signed-off-by: Joao Fernandes <joao.fernandes@docker.com>

### CODE REVIEW

File: docs/reference/commandline/run.md [hunk 1]
```diff
@@ -340,75 +340,43 @@ system's interfaces.
...
-TEST_FOO=This is a test
+$ docker run --env VAR1=value1 --env VAR2=value2 ubuntu env | grep VAR
+VAR1=value1
+VAR2=value2
 ```
 
-The `--env-file` flag takes a filename as an argument and expects each line
-to be in the `VAR=VAL` format, mimicking the argument passed to `--env`. Comment
-lines need only be prefixed with `#`
+You can also use variables that you've exported to your local environment:
```
Comment (mstanleyjones) [inline docs/reference/commandline/run.md]: I would caveat here that this is depending on your local environment and that it's less predictable than specifying the vars on the  command line or in the Dockerfile.

File: docs/reference/commandline/run.md [hunk 2]
```diff
@@ -340,75 +340,43 @@ system's interfaces.
...
+```bash
+export VAR1=value1
+export VAR2=value2
+
+$ docker run --env VAR1 --env VAR2 ubuntu env | grep VAR
+VAR1=value1
+VAR2=value2
+```
+
+If no `=` is provided and that variable is not exported in your local
```
Comment (mstanleyjones) [inline docs/reference/commandline/run.md]: Are you sure it won't be set to an empty value?

File: docs/reference/commandline/run.md [hunk 3]
```diff
@@ -340,75 +340,43 @@ system's interfaces.
...
-This sets simple (non-array) environmental variables in the container. For
-illustration all three
-flags are shown here. Where `-e`, `--env` take an environment variable and
-value, or if no `=` is provided, then that variable's current value, set via
-`export`, is passed through (i.e. `$MYVAR1` from the host is set to `$MYVAR1`
-in the container). When no `=` is provided and that variable is not defined
-in the client's environment then that variable will be removed from the
-container's list of environment variables. All three flags, `-e`, `--env` and
-`--env-file` can be repeated.
+The `-e`, `--env`, and `--env-file` flags allow you to set simple (non-array)
```
Comment (mstanleyjones) [inline docs/reference/commandline/run.md]: I like a more active sentence like:

Use the `-e`, `--env`, and `--env-file` flags to set simple (non-array) environment variables within the container.

**Review by mdlinville — CHANGES_REQUESTED** [review]
Just a few suggestions. It might be good to also visit the Dockerfile reference and make sure its info about environment variables is consistent.

Have you tested how environment variables work on Windows containers? It might be good to show an example of them in action and how they manifest within the container.

File: docs/reference/commandline/run.md (hunk 2 above)
Comment (joaofnfernandes) [inline docs/reference/commandline/run.md]: Yup, it won't be set

```
docker run -e VAR1 ubuntu env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=e5c77d8efe0a
HOME=/root
```

File: docs/reference/commandline/run.md (hunk 1 above)
Comment (joaofnfernandes) [inline docs/reference/commandline/run.md]: I included a small caveat after the snippet. I agree that you should make the declaration of variables as explicit as possible, and I bet most uses for setting variables from the environment where for situations that are now covered by secrets.

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Comment by joaofnfernandes** [comment]
@mstanleyjones I've added another caveat explaining that `docker run --env` allows you to define new environment variables, or overwrite the ones that are defined in the Dockerfile for the image you're using.

I think the Dockerfile reference is OK for now. Since the `ENV` keyword in the Dockerfile works at buildtime and `docker run --env` works at runtime, I think most people will intuitively expect the be current behavior of `docker build` and `docker run`, so I don't think we need to add more disclaimers.

I don't have a Windows machine to test with, but I think the behavior of the builder and runner should be exactly the same. You'll only have to update the samples to run the equivalent of `env` in windows, to print the environment variables that are set in the container.

Does this satisfy all your requirements to approve this PR?

**Review by mdlinville — APPROVED** [review]
LGTM. We should go back through these at some point and make it so that people don't have to already know what `cat` and `grep` do. It might be worthwhile to make the statement, then explain what the example is doing, then show the example.

**Comment by joaofnfernandes** [comment]
Agree, and also add the equivalent snippets for windows.

**Comment by rneu31** [comment]
Is there an example of --env working on Windows at all? Seems Windows was brushed over here. If I run a `docker run -it some-image:1.0 -e somevar=hello cmd` and then an `echo %somevar%` the var doesn't seem to be set.

**Pull request merged**